frame/support/procedural/src/construct_runtime/expand/call.rs

```diff
@@ -30,21 +30,25 @@ pub fn expand_outer_dispatch(
 	let mut query_call_part_macros = Vec::new();
 	let mut pallet_names = Vec::new();
 
-	let pallets_with_call = pallet_decls
-		.iter()
-		.filter(|decl| decl.exists_part("Call"));
+	for pallet_decl in pallet_decls {
+		let name = &pallet_decl.name;
 
-	for pallet_declaration in pallets_with_call {
-		let name = &pallet_declaration.name;
-		let path = &pallet_declaration.path;
-		let index = pallet_declaration.index;
+		if pallet_decl.exists_part("Call") {
+			let path = &pallet_decl.path;
+			let index = pallet_decl.index;
 
-		variant_defs.extend(quote!(#[codec(index = #index)] #name( #scrate::dispatch::CallableCallFor<#name, #runtime> ),));
-		variant_patterns.push(quote!(Call::#name(call)));
-		pallet_names.push(name);
-		query_call_part_macros.push(quote! {
-			#path::__substrate_call_check::is_call_part_defined!(#name);
-		});
+			variant_patterns.push(quote!(Call::#name(call)));
+			variant_defs.extend(quote!(#[codec(index = #index)] #name( #scrate::dispatch::CallableCallFor<#name, #runtime> ),));
+			pallet_names.push(name);
+			query_call_part_macros.push(quote! {
+				#path::__substrate_call_check::is_call_part_defined!(#name);
+			});
+		} else {
+			let deprecation_note = format!("`{}` does not have the `Call` part imported in \
+				`construct_runtime`, perhaps you have forgotten to include it?", name);
+
+			variant_defs.extend(quote!(#[deprecated = #deprecation_note] #name,));
```

Comment:
Can we just emit a warning here directly?

Reply:
If we cannot do that, can we introduce a custom lint and use it?
If we cannot do that here is the list of lints that trigger warnings by default:
https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html
Maybe one of the `unused_*` lints would be a better fit?
Like [`unused_must_use`](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#unused-must-use)?

Reply:
The objective here is to emit a warning whenever this particular variant is referenced/used, so if we use `unused_must_use`, the lint will be eliminated if the enum variant is assigned to a variable, since that counts as a "use" and so it is not "unused" anymore.

I do like the idea of writing a custom lint, but from my experience in servo, that requires writing a compiler plugin, which means we'd need to compile with rustc nightly only since compiler plugins is a nightly-only feature. I may be wrong on this as I don't know if the times have changed, but I'll do some more research to see what's possible.